AirportTicketBookingSystem/AirportTicketBookingSystem/Repositories/FlightRepository.cs

```diff
@@ -0,0 +1,175 @@
+using AirportTicketBookingSystem.Utilities;
+
+namespace AirportTicketBookingSystem.Repositories
+{
+    public class FlightRepository : IFlightRepository
+    {
+        private readonly string _filePath;
+        private readonly IBookingRepository _bookingRepository;
+
+        public FlightRepository(string filePath)
+        {
+            _filePath = filePath;
+        }
+
+        public Flight GetFlightById(int id)
+        {
+            var flightRecords = GetAllFlights();
+            var selectedFlight = flightRecords.FirstOrDefault(b => b.Id == id);
+
+            return selectedFlight ?? throw new KeyNotFoundException($"Flight with Id {id} was not found");
+        }
+
+        public void DeleteFlight(int id)
+        {
+            var flightRecords = GetAllFlights();
+            var selectedFlight = flightRecords.FirstOrDefault(b => b.Id == id);
+
+            if (selectedFlight == null)
+                throw new KeyNotFoundException($"Flight with Id {id} was not found");
+
+            flightRecords.Remove(selectedFlight);
+
+            Save(flightRecords);
+        }
+
+        public List<Flight> GetAllFlights(string? filePath = null)
+        {
+            var pathToUse = filePath ?? _filePath;
+
+            if (!File.Exists(pathToUse))
+                throw new FileNotFoundException("CSV file not found", pathToUse);
+
+            using var reader = new StreamReader(pathToUse);
+            using var csv = new CsvReader(reader, new CsvConfiguration(CultureInfo.InvariantCulture)
+            {
+                PrepareHeaderForMatch = args => args.Header.Trim(),
+                MissingFieldFound = null
+            });
+
+            var flights = csv.GetRecords<Flight>().ToList();
+
+            return flights;
+        }
+
+        public void AddFlight(Flight flight)
+        {
+            if (flight == null)
+                throw new ArgumentNullException(nameof(flight));
+
+            var records = GetAllFlights();
+            flight.Id = IdGenerator.GenerateNewId(records);
+
+            records.Add(flight);
+            Save(records);
+        }
+
+        public void UpdateFlight(int flightId, Flight newFlight)
+        {
+            if (newFlight is null) throw new ArgumentNullException(nameof(newFlight));
+
+            var records = GetAllFlights();
+            var index = records.FindIndex(f => f.Id == flightId);
+
+            if (index < 0)
+                throw new KeyNotFoundException("Flight not found to update");
+
+            records[index] = newFlight;
+            Save(records);
+        }
+
+        private void Save(List<Flight> flights)
+        {
+            using var writer = new StreamWriter(_filePath);
+            using var csv = new CsvWriter(writer, CultureInfo.InvariantCulture);
+            csv.WriteRecords(flights);
+        }
+
+        public List<Flight> SearchAvailableFlights(string departureCountry = null, string destinationCountry = null,
```

Comment:
you would need a service method to call this repo, if you have an api to consume this method, you won't call the repo directly from the API layer, you would do it through a service. Lets do it with any other similar methods.